tests/CMakeLists.txt uses target_sources but says it works with CMake 2.8.1

*Please use this template for reporting suspected bugs or requests for help.*

# Issue description

`tests/CMakeLists.txt` uses CMake's `target_sources`, which was only added in CMake 3.1. 
`cmake_minimum_required` is set to 2.8.1 in that file, and 2.8.12 in the project root.

This is causing libzmq project's Jenkins to fail:
https://libzmq.cloudapp.net/job/libzmq/1820/console

Note Jenkins isn't reporting the problem, but it's not exactly working either...

# Environment

* libzmq version (commit hash if unreleased): d5e4319edcc984e4b03f16249378ae9eca67fc01
* OS: all
* CMake: before v3.1

# Minimal test code / Steps to reproduce the issue

1.  kick off CI on Jenkins

# What's the actual result? (include assertion message & call stack if applicable)

```
CMake Error at tests/CMakeLists.txt:184 (target_sources):
  Unknown CMake command "target_sources".
```

# What's the expected result?

Jenkins should build, not just flake out and say "SUCCESS!" (Fixing Jenkins itself is a separate issue I guess...?)

--- a/builds/cmake/platform.hpp.in
+++ b/builds/cmake/platform.hpp.in
@@ -24,6 +24,8 @@
 #cmakedefine ZMQ_HAVE_SO_PEERCRED
 #cmakedefine ZMQ_HAVE_LOCAL_PEERCRED
 
+#cmakedefine ZMQ_HAVE_O_CLOEXEC
+
 #cmakedefine ZMQ_HAVE_SOCK_CLOEXEC
 #cmakedefine ZMQ_HAVE_SO_KEEPALIVE
 #cmakedefine ZMQ_HAVE_TCP_KEEPCNT

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -147,7 +147,20 @@ link_libraries(libzmq ${OPTIONAL_LIBRARIES})
 include_directories("${CMAKE_SOURCE_DIR}/../include")
 
 foreach(test ${tests})
-  add_executable(${test} ${test}.cpp)
+  # target_sources not supported before CMake 3.1
+  if (ZMQ_HAVE_CURVE AND ${test} MATCHES test_security_curve)
+    add_executable(${test} ${test}.cpp
+        "../src/tweetnacl.c"
+        "../src/err.cpp"
+        "../src/random.cpp"
+        "../src/clock.cpp"
+        "testutil_security.hpp")
+  elseif (${test} MATCHES test_security_zap)
+    add_executable(${test} ${test}.cpp
+        "testutil_security.hpp")
+  else ()
+    add_executable(${test} ${test}.cpp)
+  endif ()
   if(WIN32)
     # This is the output for Debug dynamic builds on Visual Studio 6.0
     # You should provide the correct directory, don't know how to do it automatically
@@ -179,21 +192,11 @@ if(ZMQ_HAVE_CURVE)
   set_tests_properties(test_security_curve PROPERTIES TIMEOUT 60)
 endif()
 
-#add additional required files
+#add additional required flags
 if(ZMQ_HAVE_CURVE)
-  target_sources(test_security_curve PRIVATE 
-	"../src/tweetnacl.c" 
-	"../src/err.cpp" 
-	"../src/random.cpp" 
-	"../src/clock.cpp"
-	"testutil_security.hpp"
-	)
   target_compile_definitions(test_security_curve PRIVATE "-DZMQ_USE_TWEETNACL")
 endif()
 
-target_sources(test_security_zap PRIVATE 
-  "testutil_security.hpp"
-)
 set_tests_properties(test_security_zap PROPERTIES TIMEOUT 60)
 
 #Check whether all tests in the current folder are present

